queries.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# Given a node_id for a discussion comment that is a reply in thread, return the parent comment's node ID.
+function getDiscussionReplyToId() {
+  local NODE_ID=$1
+  local REPLY_TO_DATA=$(gh api graphql -f query='
+  query($nodeId: ID!) {
+    node(id: $nodeId) {
+      ... on DiscussionComment {
+        replyTo {
+          id
+        }
+      }
+    }
+  }' -F nodeId=$NODE_ID)
+  echo $REPLY_TO_DATA | jq -r '.data.node.replyTo.id'
+}
+
+# Given a discussion node ID, a message, and an optional reply to node ID, adds a discussion comment.
+function addDiscussionComment() {
+  local DISCUSSION_NODE_ID=$1
+  local MESSAGE=$2
+  local REPLY_TO_ID=$3
+
+  if [ -n "$REPLY_TO_ID" ]; then
+    gh api graphql -F discussionId="$DISCUSSION_NODE_ID" -F replyToId="$REPLY_TO_ID" -F body="$MESSAGE" -f query='
```

Comment:
Sorry! I meant could you do something like this:

```bash
if [ -n $REPLY_TO_ID ]; then; else REPLY="$REPLY_TO_ID"; fi
gh api graphql -F discussionId="$DISCUSSION_NODE_ID" -F replyToId=$REPLY -F body="$MESSAGE" -f query='
      mutation($discussionId: ID!, , $replyToId: ID, $body: String!) {
          addDiscussionComment(input: {discussionId: $discussionId, replyToId: $replyToId, body: $body}) {
          comment {
              id
          }
        }
      }
    '
```

^ I think the var can be passed in as `null` on graphql

Reply:
note: the if statement only is used to add `""` around `REPLY_TO_ID` if it is not null.

Reply:
Oh I see. I think I tried this before and ran into an error. Let me try again to confirm.

Reply:
We run into the following error, so I think we need to stick with two separate calls.

```bash
gh: Could not resolve to a node with the global id of ''
{"data":{"addDiscussionComment":null},"errors":[{"type":"NOT_FOUND","path":["addDiscussionComment"],"locations":[{"line":3,"column":9}],"message":"Could not resolve to a node with the global id of ''"}]}
```

Reply:
Oh weird it auto resolves it to an empty string.